src/structurizr/model/component.py

```diff
@@ -108,12 +109,19 @@ def __init__(
         self.tags.add(Tags.COMPONENT)
 
     @classmethod
-    def hydrate(cls, component_io: ComponentIO, container: "Container") -> "Component":
-        """Create and hydrate a new `Component` instance from its IO."""
-        return cls(
+    def hydrate(
+        cls, component_io: ComponentIO, container: "Container", model: "Model"
+    ) -> "Component":
+        """Create and hydrate a new `Component` instance from its IO.
+
+        This will also automatically register with the model.
+        """
+        component = cls(
             **cls.hydrate_arguments(component_io),
             parent=container,
             technology=component_io.technology,
             # TODO: code_elements=map(CodeElement.hydrate, component_io.components),
             size=component_io.size,
         )
+        model += component
+        return component
```

Comment:
I'm not convinced that it is a good choice to pass in the model to the hydrate class method. That makes it impossible to hydrate only this Component from JSON should one wish to do so for debugging purposes, for example. It would be better in my opinion to have this happen in some higher up logic. I can't pin point where exactly but I guess some kind of `.add` method.

Reply:
Yeah - I was trying to establish a consistent pattern across `SoftwareSystem`, `Container`, `Component` and `DeploymentNode`.  A couple of alternative approaches:
- In `Model`, when a `SoftwareSystem` or `DeploymentNode` is added then it `Model` could explicitly add all the children.  This would be made easier if we added a `children` property to `ModelItem` which by default returned an empty list but could be overridden by subtypes (e.g. `SoftwareSystem` would return its containers, etc.).
- In `SoftwareSystem`, etc. we override `set_model` (from `ModelRefMixin`) to also set the model in its containers, etc.

I'm leaning toward the first approach (including the generic `children`) as it feels less fragile.

Reply:
I will have to dive more deeply into the code again but what you suggest sounds reasonable. I've opened an issue for it so that we can proceed with this PR.